### PR TITLE
 Move stats to metrics

### PR DESCRIPTION
**Pull request by bogdandrutu**
The main idea is to reduce confusion about why two top level APIs exists to record metrics/stats/measurement.

We still need to offer the ability to record raw measurements as well as pre-defined metrics (including aggregation and labels).

The structure of the Metrics package is not final:

Do we need MetricsRegistry or we move all the methods to Meter? The SDK will allow users to set Filters that apply to all created metrics (similar with metric registry in Micrometer for example).
The Views API in the SDK is not yet moved to metrics.

This PR is based on https://github.com/open-telemetry/opentelemetry-java/pull/226 which was started from a branch that lived in the main repo because that was under my account. We will move all the discussions here.

### CODE REVIEW

**Comment by bogdandrutu** [comment]
@yurishkuro commented:
Some questions/comments:
  * in many other metrics APIs that allow recording gauges, timers, and/or histograms, the recording API is still about recording the _raw measure_, but behind the scene the library may or may not perform some form of aggregation, like `LastValue` for gauges, or `Histogram` for timers. So the distinction between raw measure and Gauge/Histogram is not clear to me, since from instrumentation point of view it's always recording  the raw value.
  * on the other hand, counters seem to be fundamentally different and can never be captured via raw measure, is that correct? This breaks the mental model for me.
  * being able to change aggregation without touching instrumentation is great - I often wanted to see something like queue_length not as a gauge but as a histogram (incidentally tally didn't support histograms for non-timer values)
  * application of request-scoped tags is also very tricky. As @bogdandrutu said, there are some stats that don't make sense to be labeled with request-scoped tags, e.g. CPU utilization. For measures where we do want to apply context tags, the ability to define views also provides a place to define context tag rules, potentially on a per-measure basis, e.g. "allow tags A and B for measureX, but only tag B for measure Y" (however, see next item).
  * the requirement to define views seems very burdensome, especially because they need to reference the actual measure objects. In a complex, multi-layer system like Jaeger backend, the metrics defined by the individual components are usually encapsulated within those components, and it's not feasible to have a single place in the app where I can define views for all measures from all components. On the other hand, if views are defined internally by the components themselves, then it mostly makes the views redundant, since the components can define aggregations at the time of constructing the measures, which is what typically happens with most other metrics APIs where you explicitly ask for Counter, Gauge, Histogram. I'd be interested to see how more complex systems using OC deal with this encapsulation issue.

**Comment by SergeyKanzhelev** [comment]
>  * in many other metrics APIs that allow recording gauges, timers, and/or histograms, the recording API is still about recording the _raw measure_, but behind the scene the library may or may not perform some form of aggregation, like `LastValue` for gauges, or `Histogram` for timers. So the distinction between raw measure and Gauge/Histogram is not clear to me, since from instrumentation point of view it's always recording  the raw value.
>  * on the other hand, counters seem to be fundamentally different and can never be captured via raw measure, is that correct? This breaks the mental model for me.

So this PR doesn't make anything worse, correct? We might need an issue for this. It is a good property to be able to express the `SpanData` of metrics. Basically read from file and proxy the metric.


>  * application of request-scoped tags is also very tricky. As @bogdandrutu said, there are some stats that don't make sense to be labeled with request-scoped tags, e.g. CPU utilization. For measures where we do want to apply context tags, the ability to define views also provides a place to define context tag rules, potentially on a per-measure basis, e.g. "allow tags A and B for measureX, but only tag B for measure Y" (however, see next item).
  * the requirement to define views seems very burdensome, especially because they need to reference the actual measure objects. In a complex, multi-layer system like Jaeger backend, the metrics defined by the individual components are usually encapsulated within those components, and it's not feasible to have a single place in the app where I can define views for all measures from all components. On the other hand, if views are defined internally by the components themselves, then it mostly makes the views redundant, since the components can define aggregations at the time of constructing the measures, which is what typically happens with most other metrics APIs where you explicitly ask for Counter, Gauge, Histogram. I'd be interested to see how more complex systems using OC deal with this encapsulation issue.

Couple of thoughts - I think tags in views is an extremely powerful tool for customer defined tags. Furthermore for some metrics like `http.response_time` - they should be associated with span itself (or parent span or root span), rather than with the tags that supposed to be set on the context. Tags may be left for the customer defined properties.

CPU metrics will never be reported from the thread that has tags. So there will never be a case when request properties got applied to global CPU perf counter.

Again, perhaps we need to start filing individual issues for this feedback, as it touches some interesting points.

File: sdk/src/main/java/openconsensus/sdk/stats/View.java [hunk 1]
```diff
@@ -23,7 +23,7 @@
 import javax.annotation.concurrent.Immutable;
 import openconsensus.internal.StringUtils;
 import openconsensus.internal.Utils;
-import openconsensus.stats.Measure;
```
Comment (SergeyKanzhelev) [inline sdk/src/main/java/openconsensus/sdk/stats/View.java]: so you kept tests namespace for now?

Comment (bogdandrutu) [reply sdk/src/main/java/openconsensus/sdk/stats/View.java]: This is in the SDK, I did not change that for the moment.

**Comment by yurishkuro** [comment]
@SergeyKanzhelev I broke my comments into two issues: #268 and #269.

**Pull request merged**